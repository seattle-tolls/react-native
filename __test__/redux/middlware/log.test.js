import logMiddleware from '../../../src/redux/middleware/log'

describe('MIDDLEWARE', ()=>{
  describe('middleware/log', ()=>{
    beforeEach(()=> {
      next = jest.fn()
      dispatch = jest.fn()
      getState = jest.fn()
      middleware = logMiddleware({ dispatch, getState})(next)
    })

    it('should process the log fn', ()=>{
      const sampleAction = { type: 'LOGGING_SOMETHING'}
      middleware(sampleAction)
      expect(next.mock.calls).toMatchSnapshot()
    })
  })
})