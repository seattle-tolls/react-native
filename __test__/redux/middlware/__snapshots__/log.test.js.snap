// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOG MIDDLEWARE middleware/log should process the log fn 1`] = `
Array [
  Array [
    Object {
      "type": "SAMPLE_ACTION",
    },
  ],
]
`;
